cmd/bosun/expr/expr.go

```diff
@@ -212,6 +212,20 @@ func (r ResultSlice) DescByValue() ResultSlice {
 	return c
 }
 
+// Filter strips out results if their tagset doesn't conform to the given key/value pair restrictions
+// It updates the input slice in place.
+func (r ResultSlice) Filter(filter opentsdb.TagSet) ResultSlice {
+	lastGoodIndex := -1
+
+	for _, res := range r {
+		if res.Group.Compatible(filter) {
+			lastGoodIndex++
+			r[lastGoodIndex] = res
```

Comment:
it seemed messy (and slower) to allocate a new array, as far as i see there's no reason to do this.
this is also how sort.Sort works.
is there a good reason to allocate a new array?